dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -370,9 +370,24 @@ private Container getContainer(final String containerId, final User user, final
 
         if (FileAssetContainerUtil.getInstance().isFolderAssetContainerId(containerId)) {
 
-            return mode.showLive?
-                    this.containerAPI.getLiveContainerByFolderPath   (containerId, host, user, mode.respectAnonPerms):
-                    this.containerAPI.getWorkingContainerByFolderPath(containerId, host, user, mode.respectAnonPerms);
+            final Optional<Host> hostOpt = HostUtil.getHostFromPathOrCurrentHost(containerId, Constants.CONTAINER_FOLDER_PATH);
+            final Host   containerHost   = hostOpt.isPresent()? hostOpt.get():host;
+            final String relativePath    = FileAssetContainerUtil.getInstance().getPathFromFullPath(containerHost.getHostname(), containerId);
+            try {
+                
+                return mode.showLive ?
+                        this.containerAPI.getLiveContainerByFolderPath(relativePath, containerHost, user, mode.respectAnonPerms) :
+                        this.containerAPI.getWorkingContainerByFolderPath(relativePath, containerHost, user, mode.respectAnonPerms);
+            } catch (NotFoundInDbException e) {
+
+                // if does not found in the host path or current host, tries the default one if it is not the same
+                final Host defaultHost = WebAPILocator.getHostWebAPI().findDefaultHost(user, false);
+                if (!defaultHost.getIdentifier().equals(containerHost.getIdentifier())) {
+
+                    return  this.containerAPI.getWorkingContainerByFolderPath(relativePath,
```

Comment:
Are you sure that this call to `containerAPI.getWorkingContainerByFolderPath` will not throw an exception?

Reply:
it could throw an exception and it would be expected, the handlers will catch them